sources/packages/web/src/services/http/EducationProgramApi.ts

```diff
@@ -49,15 +54,41 @@ export class EducationProgramApi extends HttpBaseClient {
     }
   }
 
+  /**
+   * Prepare the API to fetch all institution
+   * location program.
+   * @param locationId location id
+   * @param page, page number if nothing is passed then
+   * DEFAULT_PAGE_NUMBER is taken
+   * @param pageLimit, limit of the page if nothing is
+   * passed then DEFAULT_PAGE_LIMIT is taken
+   * @param searchName,program name keyword to be searched
+   * @param sortField, field to be sorted
+   * @param sortOrder, order to be sorted
+   * @returns program summary for an institution location.
+   */
   public async getLocationProgramsSummary(
     locationId: number,
-  ): Promise<SummaryEducationProgramDto[]> {
+    page = DEFAULT_PAGE_NUMBER,
+    pageCount = DEFAULT_PAGE_LIMIT,
+    searchProgramName?: string,
+    sortField?: InstitutionProgramSummaryFields,
+    sortOrder?: DataTableSortOrder,
+  ): Promise<EducationProgramsSummaryPaginated> {
     try {
-      const response = await this.apiClient.get(
-        `institution/education-program/location/${locationId}/summary`,
-        this.addAuthHeader(),
-      );
-      return response.data as SummaryEducationProgramDto[];
+      let URL = `institution/education-program/location/${locationId}/summary?page=${page}&pageLimit=${pageCount}`;
+      if (searchProgramName) {
+        URL = `${URL}&searchProgramName=${searchProgramName}`;
+      }
+      if (sortField && sortOrder) {
+        const sortDBOrder =
+          sortOrder === DataTableSortOrder.DESC
```

Comment:
Do we need a _DataTableSortOrder_ and a _FieldSortOrder_? If both are just ASC/DESC can we use only the FieldSortOrder that seems to have a more generic name?

Reply:
Datatable sort order is 1,-1 but in typeorm it is "ASC","DESC"

Reply:
Thanks @dheepak-aot I did not noticed it 😉